internal/service/auth/auth.go

```diff
@@ -0,0 +1,229 @@
+package auth
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/google/uuid"
+	db2 "github.com/kamencov/go-musthave-diploma-tpl/internal/storage/db"
+	"time"
+
+	"github.com/kamencov/go-musthave-diploma-tpl/internal/customerrors"
+	"github.com/kamencov/go-musthave-diploma-tpl/internal/service"
+	"github.com/kamencov/go-musthave-diploma-tpl/internal/service/auth/entity"
+)
+
+//go:generate mockgen -source=./auth.go -destination=/Users/pavel/GolandProjects/go-musthave-diploma-tpl/internal/mocks/mock_storage_auth.go -package=mocks
+type AuthService interface {
+	RegisterUser(ctx context.Context, login, password string) error
+	AuthUser(ctx context.Context, login, password string) (entity.Tokens, error)
+	VerifyUser(token string) (string, error)
+	RefreshToken(ctx context.Context, token string) (entity.Tokens, error)
+	GeneratedTokens(ctx context.Context, login string) (entity.Tokens, error)
+	HashPassword(password string) string
+}
+
+type ServiceAuth struct {
+	tokenSalt    []byte
+	passwordSalt []byte
+
+	accessTokenTTL  time.Duration
+	refreshTokenTTL time.Duration
+
+	storage service.Storage
+}
+
+func NewServiceAuth(saltTKN, saltPSW []byte, storage service.Storage) *ServiceAuth {
+	return &ServiceAuth{
+		tokenSalt:    saltTKN,
+		passwordSalt: saltPSW,
+
+		accessTokenTTL:  24 * time.Hour,
+		refreshTokenTTL: 24 * time.Hour,
+
+		storage: storage,
+	}
+}
+
+func (s *ServiceAuth) RegisterUser(ctx context.Context, login, password string) error {
+	err := s.storage.CheckTableUserLogin(ctx, login)
+
+	if errors.Is(err, customerrors.ErrUserAlreadyExists) {
+		return customerrors.ErrUserAlreadyExists
+	}
+
+	if err != nil {
+		return err
+	}
```

Comment:
Стоит сначала проверить, что ошибка не `nil`.
```suggestion
	if err != nil {
                        if errors.Is(err, customerrors.ErrUserAlreadyExists) {
                                  return customerrors.ErrUserAlreadyExists
                        }
			return err
	}
```